### PR TITLE
Update golang

### PR DESCRIPTION
**Pull request by jayanthvn**
**What type of PR is this?**
Update golang 1.16+ to fix lumberjack dependency issue.

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:

**Does this change require updates to the CNI daemonset config files to work?**:

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jayanthvn** [comment]
Build is working fine now.

**Comment by orsenthil** [comment]
Sounds good. Excellent.

**Comment by a7i** [comment]
Are there any plans to cherry-pick this to 1.9 release?

File: go.mod [hunk 1]
```diff
@@ -19,10 +19,12 @@ require (
 	github.com/vishvananda/netlink v1.1.1-0.20201029203352-d40f9887b852
 	go.uber.org/zap v1.15.0
 	golang.org/x/lint v0.0.0-20210508222113-6edffad5e616 // indirect
-	golang.org/x/net v0.0.0-20210614182718-04defd469f4e
```
[the remark this answers is not in the document]
Comment (jayanthvn) [reply go.mod:22]: I think it is fine to keep it go 1.14 Ref : https://github.com/golang/go/issues/30791#issuecomment-472217112. But there are some proposals around this https://github.com/golang/go/issues/4613. Will look into it further and if needed will take it up in the following PR.

**Pull request merged**